### PR TITLE
K8SPG-112 Add tls auto handling

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| upgrade | passed |
| demand-backup | passed |
| smart-update | passed |
| scaling | passed |
| version-service | passed |
| recreate | passed |
| affinity | passed |
| scheduled-backup | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| users | passed |
| ns-mode | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/17ddc49becdc108cbbf5a117153c10fc8fa4eb38
image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-17ddc49b-pgo-deployer`

**Review by egegunes — COMMENTED** [review]
I have a small comment. Also please make the linter happy

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | passed |
| upgrade | passed |
| smart-update | passed |
| scaling | passed |
| version-service | passed |
| recreate | passed |
| scheduled-backup | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| users | passed |
| ns-mode | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/4743d178d3995dc23ece4333b3601e375fd61760
image: `perconalab/percona-postgresql-operator:PR-199-4743d178-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-4743d178-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-4743d178-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-4743d178-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-4743d178-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-4743d178-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/ba437d6f92424d132a9eb4753cb0d39ad35e25be
image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-ba437d6f-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/55c58a4d29778b9bebf573578779461bc8620b37
image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/55c58a4d29778b9bebf573578779461bc8620b37
image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-55c58a4d-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/f3031ed9fd27165ff618786f992be0e2bed6e950
image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-f3031ed9-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/e0beccf151cc5adb0b6b42444392e40e93c43955
image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-e0beccf1-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/97f05063d6d3acc395d332ae6a1fd9fde67a9b1a
image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade | passed |
| demand-backup | passed |
| init-deploy | passed |
| smart-update | passed |
| scaling | passed |
| version-service | passed |
| recreate | passed |
| scheduled-backup | failed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| users | passed |
| ns-mode | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/385ba493f5472dddc7fe2bdf961fe0c29d64a542
image: `perconalab/percona-postgresql-operator:PR-199-385ba493-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-385ba493-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-385ba493-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-385ba493-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-385ba493-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-385ba493-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/97f05063d6d3acc395d332ae6a1fd9fde67a9b1a
image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| init-deploy | passed |
| demand-backup | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| scheduled-backup | passed |
| monitoring | passed |
| self-healing | passed |
| operator-self-healing | passed |
| clone-cluster | passed |
| tls-check | passed |
| users | passed |
| ns-mode | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/97f05063d6d3acc395d332ae6a1fd9fde67a9b1a
image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-97f05063-pgo-deployer`

File: percona/tls/tls.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		SerialNumber:          serialNumber,
+		Subject:               subject,
+		Issuer:                issuer,
+		NotBefore:             time.Now(),
+		NotAfter:              validityNotAfter,
+		DNSNames:              hosts,
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		BasicConstraintsValid: true,
+		IsCA:                  false,
```
Comment (egegunes) [inline percona/tls/tls.go:74]: We're setting `isCA: true` if we create the key pair using cert-manager in https://github.com/percona/percona-postgresql-operator/blob/97f05063d6d3acc395d332ae6a1fd9fde67a9b1a/percona/controllers/pgc/tls.go#L162

Comment (nonemax) [reply percona/tls/tls.go:74]: https://github.com/percona/percona-postgresql-operator/pull/199/commits/2493db43abbe08183d7a8c34001a3579b5899ef7

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/f66aeb17f3e743e74c83bc0e45bf7f2e55dbf962
image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/199/commits/f66aeb17f3e743e74c83bc0e45bf7f2e55dbf962
image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-199-f66aeb17-pgo-deployer`

**Pull request merged**